hs/src/Reach/Parser.hs

```diff
@@ -142,61 +158,69 @@ instance Pretty JSBundle where
           ("// " <> viaShow rs) :
           map (pretty . ppShow) jms
 
-gatherDeps_fc :: SrcLoc -> IORef JSBundleMap -> JSFromClause -> IO JSFromClause
-gatherDeps_fc at fmr (JSFromClause ab aa s) = do
-  s_abs <- gatherDeps_file GatherNotTop (srcloc_at "import from" (tp ab) at) fmr $ trimQuotes s
-  return $ JSFromClause ab aa s_abs
 
-gatherDeps_imd :: SrcLoc -> IORef JSBundleMap -> JSImportDeclaration -> IO JSImportDeclaration
-gatherDeps_imd at fmr = \case
+gatherDeps_fc :: AppT JSFromClause
+gatherDeps_fc (JSFromClause ab aa s) = do
+  local (\e -> e { srcloc = srcloc_at "import from" (tp ab) (srcloc e) }) $ do
+    s_abs <- gatherDeps_file GatherNotTop $ trimQuotes s
+    return $ JSFromClause ab aa s_abs
+
+gatherDeps_imd :: AppT JSImportDeclaration
+gatherDeps_imd = \case
   JSImportDeclaration ic fc sm -> do
-    fc' <- gatherDeps_fc at fmr fc
+    fc' <- gatherDeps_fc fc
     return $ JSImportDeclaration ic fc' sm
   JSImportDeclarationBare a s sm -> do
-    s_abs <- gatherDeps_file GatherNotTop (srcloc_at "import bare" (tp a) at) fmr $ trimQuotes s
-    return $ JSImportDeclarationBare a s_abs sm
+    local (\e -> e { srcloc = srcloc_at "import bare" (tp a) (srcloc e) }) $ do
```

Comment:
Eval has a function that is like this local called `withAt`

Reply:
Just want to make sure: for `expect_` and `withAt`, are you saying I should _import_ those or create new ones in `Parser.hs` and `ImportSource.hs`?

Trying to import causes cyclic dependency errors. I renamed `expect_thrown'` to `expect_` in my last commit but I'm not certain that's what you're asking for.

Reply:
I mean that you should define these functions yourself and follow the "pattern" of the rest of the code. The types aren't the same, so the exact functions can't exactly work

Reply:
Thanks